src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,19 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Machine> {
+
+    @Override
+    public List<Machine> get() {
+        List<Machine> trucks = new ArrayList<>();
```

Comment:
TruckProducer should return the list of only Trucks. If we have Machine list we could put there other machine children and TruckProducer will return list which contains not only trucks. And this will be wrong work of TruckProducer.